### PR TITLE
3279 generic email config

### LINKED ISSUE #3279: Add erasure/consent email config to SaaS config
### Is your feature request related to a specific problem?

_A description of what the problem is._
_Ex. I'm always frustrated when [...]_

### Describe the solution you'd like

_A description of what you want to happen._

### Describe alternatives you've considered, if any

_A description of any alternative solutions or features you've considered._

### Additional context

_Add any other context or screenshots about the feature request here._

### PR DESCRIPTION
**Pull request by galvana**
Closes #3279 

### Code Changes

* [ ] Adding two new connection types `generic_erasure_email` and `generic_consent_email`
* [ ] Using existing `EmailSchema` for generic erasure email connectors
* [ ] Using existing `ExtendedEmailSchema` for generic consent email connectors

### Steps to Confirm

* [ ] Duplicated existing Attentive and Sovrn tests but implemented those connectors as generic erasure and consent connectors

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

File: clients/admin-ui/src/features/datastore-connections/constants.ts [hunk 1]
```diff
@@ -34,7 +34,11 @@ export const CONNECTOR_LOGOS_PATH = "/images/connector-logos/";
  * List of connection type image key/value pairs
  */
 export const CONNECTION_TYPE_LOGO_MAP = new Map<ConnectionType, string>([
+  [ConnectionType.ATTENTIVE, "attentive.svg"],
```
Comment (galvana) [inline clients/admin-ui/src/features/datastore-connections/constants.ts:37]: Sorting alphabetically

File: clients/admin-ui/src/types/api/models/ConnectionType.ts [hunk 2]
```diff
@@ -23,4 +23,6 @@ export enum ConnectionType {
   MANUAL_WEBHOOK = "manual_webhook",
   TIMESCALE = "timescale",
   FIDES = "fides",
+  GENERIC_ERASURE_EMAIL = "erasure_email",
+  GENERIC_CONSENT_EMAIL = "consent_email",
```
Comment (galvana) [inline clients/admin-ui/src/types/api/models/ConnectionType.ts:27]: Adding two new connection types, one for each type of generic email.

File: tests/ops/integration_tests/test_integration_generic_email.py
Comment (galvana) [inline tests/ops/integration_tests/test_integration_generic_email.py:1]: Replicating the existing Attentive tests but using the generic erasure email connector

**Review by adamsachs — COMMENTED** [review]
this looks great @galvana - i like how minimal the application code changes are, and per our discussions, it feels like this _should_ get us what we need at this point. your tests are comprehensive and clear - really nice work on those. thanks for taking into account the test coverage of initial privacy request execution getting into the correct state that i'd requested.

i do want to note that currently the UI throws an error (see attached screenshot) when trying to create a connection with one of these generic connector types. this isn't wholly unexpected given what i know about the code, specifically around the `identity_type` requirements and that the UI is not providing any of that data.

i guess there are two things to confirm:
- will this issue will be resolved "naturally" as part of the ongoing UI refactor?
- is it a problem that, until the UI refactor is merged into `main`, this UX path that leads to an automatic error will be in `main`? (not sure what the schedule is for getting the UI refactor into `main`...)

image

File: src/fides/api/ctl/migrations/versions/ed46521679fb_add_generic_email_connector_types.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # Add 'generic_consent_email' and 'generic_erasure_email' to ConnectionType enum
+    op.execute("alter type connectiontype rename to connectiontype_old")
+    op.execute(
+        "create type connectiontype as enum('mongodb', 'mysql', 'https', 'snowflake', "
+        "'redshift', 'mssql', 'mariadb', 'bigquery', 'saas', 'manual', 'manual_webhook', "
```
Comment (adamsachs) [inline src/fides/api/ctl/migrations/versions/ed46521679fb_add_generic_email_connector_types.py:23]: whoops, missed this -- but based on the CI test failures, looks like we're missing at least `postgres` here from the new enum values...

**Review by adamsachs — APPROVED** [review]
lgtm, thanks for the extra care in getting this across the line!

**Pull request merged**